include/aws/http/private/h2_stream.h

```diff
@@ -64,18 +64,15 @@ struct aws_h2_stream {
     } synced_data;
 };
 
-struct aws_h2_stream;
-
-AWS_EXTERN_C_BEGIN
-
-AWS_HTTP_API
 const char *aws_h2_stream_state_to_str(enum aws_h2_stream_state state);
 
-AWS_HTTP_API
```

Comment:
should we actively hide these from the symbol exports? or does -fvisibility=hidden do this for us?

Reply:
I don't know. Linker stuff is not my strong suit.
I know public API stuff needs AWS_HTTP_API
And private-but-called from tests needs AWS_HTTP_API

These are private, not called from tests, but called from other .c files in this project. What *is* the right thing to do here?

Reply:
I'm like 99% sure on all compilers later than gcc 4, we hide all symbols not explicitly marked with the API macros

Reply:
We discussed this in chat, and leaving off the exports, like I'm doing here, seems like the right approach